File: backend/app/violation_types.py
"""Tipos de violação — lógica pura, reutilizável em rotas e testes unitários."""

VIOLATION_LABELS: dict[str, str] = {
    "racismo": "Racismo",
    "discriminacao_genero": "Discriminação de gênero",
    "discriminacao_orientacao": "Discriminação por orientação sexual",
    "discriminacao_religiosa": "Discriminação religiosa",
    "discriminacao_deficiencia": "Discriminação por deficiência",
    "violencia": "Violência",
    "servico_publico": "Serviço público (educação, saúde, segurança)",
    "outro": "Outro",
}


def tipos_violacao_payload() -> list[dict[str, str]]:
    """Lista no formato da API GET /api/denuncias/tipos-violacao."""
    return [{"id": k, "label": v} for k, v in VIOLATION_LABELS.items()]

File: spring-backend/src/main/java/com/equivoz/web/ViolationTypes.java
package com.equivoz.web;

import com.equivoz.dto.ViolationTypeDto;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

public final class ViolationTypes {

    private static final Map<String, String> LABELS = new LinkedHashMap<>();

    static {
        LABELS.put("racismo", "Racismo");
        LABELS.put("discriminacao_genero", "Discriminação de gênero");
        LABELS.put("discriminacao_orientacao", "Discriminação por orientação sexual");
        LABELS.put("discriminacao_religiosa", "Discriminação religiosa");
        LABELS.put("discriminacao_deficiencia", "Discriminação por deficiência");
        LABELS.put("violencia", "Violência");
        LABELS.put("servico_publico", "Serviço público (educação, saúde, segurança)");
        LABELS.put("outro", "Outro");
    }

    private ViolationTypes() {}

    public static List<ViolationTypeDto> asList() {
        return LABELS.entrySet().stream()
                .map(e -> new ViolationTypeDto(e.getKey(), e.getValue()))
                .toList();
    }
}
